eng/build.sh

```diff
@@ -2,63 +2,514 @@
+
+    # If Xcode 9.2 exists (like on the CI/build machines), use that. Xcode 9.3 or 
+    # greater (swift 4.1 lldb) doesn't work that well (seg faults on exit).
+    if [ -f "/Applications/Xcode_9.2.app/Contents/Developer/usr/bin/lldb" ]; then
+        if [ -f "/Applications/Xcode_9.2.app/Contents/SharedFrameworks/LLDB.framework/LLDB" ]; then
+            export LLDB_PATH=/Applications/Xcode_9.2.app/Contents/Developer/usr/bin/lldb
+            export LLDB_LIB=/Applications/Xcode_9.2.app/Contents/SharedFrameworks/LLDB.framework/LLDB
+        fi
+    fi
+
+    if [ ! -f $LLDB_LIB ]; then
+        echo "Cannot find the lldb library. Try installing Xcode."
+        exit 1
+    fi
+
+    # Workaround bad python version in /usr/local/bin/python2.7 on lab machines
+    export PATH=/usr/bin:$PATH
+    which python
+    python --version
+fi
+
+#
+# Build native components
+#
+
+if [ ! -e $__DotNetCli ]; then
+   echo "dotnet cli not installed $__DotNetCli"
+   exit 1
+fi
+
+if [ $__NativeBuild == true ]; then
+    if [[ $__CI == true ]]; then
+        echo "Generating Version Source File"
+        __GenerateVersionRestoreLog="$__LogDir/GenerateVersionRestore.binlog"
+        $__DotNetCli msbuild $__ProjectRoot/eng/CreateVersionFile.csproj /v:$__Verbosity /bl:$__GenerateVersionRestoreLog /t:Restore /p:Configuration="$__BuildType" /p:Platform="$__BuildArch" $__UnprocessedBuildArgs
+
+        __GenerateVersionLog="$__LogDir/GenerateVersion.binlog"
+        $__DotNetCli msbuild $__ProjectRoot/eng/CreateVersionFile.csproj /v:$__Verbosity /bl:$__GenerateVersionLog /t:GenerateVersionFiles /p:GenerateVersionSourceFile=true /p:NativeVersionSourceFile="$__IntermediatesDir/version.cpp" /p:Configuration="$__BuildType" /p:Platform="$__BuildArch" $__UnprocessedBuildArgs
+        if [ $? != 0 ]; then
+            echo "Generating Version Source File FAILED"
+            exit 1
+        fi
+    else
+        echo "Generating Empty Version Source File"
+       echo "" > "$__IntermediatesDir/version.cpp"
+    fi
+
+    build_native "$__BuildArch" "$__IntermediatesDir" "$__ExtraCmakeArgs"
+fi
+
+#
+# Copy the native SOS binaries to where these tools expect for testing
+#
+
+if [[ $__NativeBuild == true || $__Test == true ]]; then
+    __dotnet_sos=$__RootBinDir/bin/dotnet-sos/$__BuildType/netcoreapp2.1/publish/$__DistroRid
+    __dotnet_dump=$__RootBinDir/bin/dotnet-dump/$__BuildType/netcoreapp2.1/publish/$__DistroRid
+
+    mkdir -p "$__dotnet_sos"
+    mkdir -p "$__dotnet_dump"
+
+    cp "$__BinDir"/* "$__dotnet_sos"
+    echo "Copied SOS to $__dotnet_sos"
+
+    cp "$__BinDir"/* "$__dotnet_dump"
+    echo "Copied SOS to $__dotnet_dump"
+fi
+
+#
+# Run xunit tests
+#
+
+if [ $__Test == true ]; then
+   if [ $__CrossBuild != true ]; then
+      if [ "$LLDB_PATH" == "" ]; then
+          export LLDB_PATH="$(which lldb-3.9.1 2> /dev/null)"
+          if [ "$LLDB_PATH" == "" ]; then
+              export LLDB_PATH="$(which lldb-3.9 2> /dev/null)"
```

Comment:
Shouldn't we try to respect the clang version provided? Also would this be a better alternative?

```bash
lldb_local_version=`command -v lldb-3.9.1 || command -v lldb-3.9 || command -v lldb-4.0 || command -v lldb`
if [ -f "$lldb_local_version" ]; then
    export LLDB_PATH="$lldb_local_version"
else 
    echo "LLDB not found"
fi
```

Reply:
This logic has always been in the old build-native.sh I don't want to change what works.